### ISSUE TITLE
Documentation enhancements

### ISSUE THREAD

**Issue by Tharushan** [prebid/prebid-server#2076]
Hello,

I wanted to setup a prebid-server project using the documentation but I was confronted to missing guidelines for some of the following points : 

- First to start the prebid server, you'll need to setup `gdpr.default_value` using env variables like `PBS_GDPR_DEFAULT_VALUE` or using a pbs.yaml file 
This was also mentioned in this issue :  https://github.com/prebid/prebid-server/issues/1897

- To request bids, your website using Prebid.js and your Prebid-Server must follow theses 2 elements: 
  - The host must not be `localhost` 
  - HTTPS must be activated
 
  This is mentioned in StackOverflow answer : https://stackoverflow.com/a/49971455

- For some bidders, as it's mentioned in Prebid documentation, you'll need to contact them for approval (for exemple with Rubicon and IndexExchange) : 
  > For both Prebid.js and Prebid-Server, the Rubicon Project adapter requires setup and approval from the Magnite team, even for existing accounts. Please reach out to your account team or globalsupport@magnite.com for more information.
    
    But you will also have to set some env variables on Prebid-Server like : 
  - `​​PBS_ADAPTERS_RUBICON_XAPI_USERNAME`
  - `PBS_ADAPTERS_RUBICON_XAPI_PASSWORD`
  and maybe activate them:
  - `PBS_ADAPTERS_RUBICON_DISABLED: "false"`
  
  Are there other elements that someone who wants to setup prebid-server for the first time should know ? 
  
  Thanks

**Comment by mansinahar** [comment]
@Tharushan you're right, we do lack Prebid Server setup documentation but it's on our list of things to work on unless you'd be willing to contribute :)

I think what you have above should cover the most part of it. There are a few other configurations that you'd need based on the features you want to use and how you want to use them. For example, if you'd like to use the Stored Requests capability, Prebid Server has three ways to support that. Using file system, HTTP endpoint and database. Based on which of these three ways you want to support (you can support zero or more), you'd have to set the appropriate environment variables. For example, for file system and HTTP endpoint support, you'd have to set the following:

```
  PBS_STORED_REQUESTS_FILESYSTEM_ENABLED: "true"
  PBS_STORED_REQUESTS_FILESYSTEM_DIRECTORYPATH: "/a/path/on/your/server"
  PBS_STORED_REQUESTS_HTTP_AMP_ENDPOINT: "https://some-amp-stored-request-api.com/stored-requests"
  PBS_STORED_REQUESTS_HTTP_ENDPOINT: "https://some-stored-request-api.com/stored-requests"
  PBS_STORED_REQUESTS_IN_MEMORY_CACHE_TYPE: "unbounded" // Supported values are none, unbounded, lru
```

**Note:** Prebid Server supports in memory caching of stored requests so the in memory cache type configuration is related to that.

Please refer [this](https://github.com/prebid/prebid-server/blob/master/config/config.go#L19) config struct for more Prebid Server configuration options and if you have any questions about any of those configs or what they mean, we're happy to answer them on this thread :)
